examples/async_tasks/external_source_compute_thread_pool.rs

```diff
@@ -0,0 +1,99 @@
+use bevy::{
+    prelude::*,
+    tasks::{AsyncComputeTaskPool, Task},
+};
+use crossbeam_channel::{unbounded, Receiver};
+use futures_lite::future;
+use rand::Rng;
+
+fn main() {
+    App::new()
+        .add_event::<StreamEvent>()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup_system)
+        .add_system(read_stream_system)
+        .add_system(spawn_text_system)
+        .add_system(move_text_system)
+        .run();
+}
+
+struct StreamReceiver(Receiver<u32>);
+struct StreamTask(Task<()>);
+struct StreamEvent(u32);
+
+fn setup_system(mut commands: Commands, thread_pool: Res<AsyncComputeTaskPool>) {
+    commands.spawn_bundle(OrthographicCameraBundle::new_2d());
+
+    let (tx, rx) = unbounded::<u32>();
```

Comment:
There's no back pressure here. Obviously in this case it's not needed, but I feel like we should at least mention it, e.g. use a bounded channel of 10 instead.